hibernate-core/src/main/java/org/hibernate/dialect/DB2Dialect.java

```diff
@@ -127,47 +131,33 @@ public int getDefaultStatementBatchSize() {
 	}
 
 	@Override
-	protected String columnType(int jdbcTypeCode) {
-		if ( getDB2Version().isBefore( 11 ) ) {
-			switch (jdbcTypeCode) {
-				case BOOLEAN:
-					// prior to DB2 11, the 'boolean' type existed,
-					// but was not allowed as a column type
-					return "smallint";
-				case BINARY: // should use 'binary' since version 11
-				case VARBINARY: // should use 'varbinary' since version 11
-					return "varchar($l) for bit data";
-			}
-		}
+	protected void registerColumnTypes(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
+		super.registerColumnTypes( typeContributions, serviceRegistry );
+		final SqlTypeRegistry sqlTypeRegistry = typeContributions.getTypeConfiguration().getSqlTypeRegistry();
 
-		switch (jdbcTypeCode) {
-			case TINYINT:
-				// no tinyint
-				return "smallint";
-			case NUMERIC:
-				// HHH-12827: map them both to the same type to avoid problems with schema update
-				// Note that 31 is the maximum precision DB2 supports
-				return super.columnType(DECIMAL);
-			case BLOB:
-				return "blob($l)";
-			case CLOB:
-				return "clob($l)";
-			case TIMESTAMP_WITH_TIMEZONE:
-				return "timestamp($p)";
-			case TIME_WITH_TIMEZONE:
-				return "time";
-			default:
-				return super.columnType(jdbcTypeCode);
-		}
-	}
+		// no tinyint
+		sqlTypeRegistry.addDescriptor( new SqlTypeImpl( TINYINT, "smallint", this ) );
+		// HHH-12827: map them both to the same type to avoid problems with schema update
+		// Note that 31 is the maximum precision DB2 supports
+		sqlTypeRegistry.addDescriptor( new SqlTypeImpl( NUMERIC, "decimal($p,$s)", this ) );
+		sqlTypeRegistry.addDescriptor( new SqlTypeImpl( BLOB, "blob($l)", this ) );
+		sqlTypeRegistry.addDescriptor( new SqlTypeImpl( CLOB, "clob($l)", this ) );
+
+		sqlTypeRegistry.addDescriptor( new SqlTypeImpl( TIMESTAMP_WITH_TIMEZONE, "timestamp($p)", this ) );
+		sqlTypeRegistry.addDescriptor( new SqlTypeImpl( TIME_WITH_TIMEZONE, "time", this ) );
```

Comment:
Wait, what, what's going on here?

After I went to rather a lot of effort to get rid of all these verbose registrations and replace them by much nicer `switch` statements, you're now rolling all that back?!